azext_iot/iothub/providers/state.py

```diff
@@ -0,0 +1,692 @@
+                        )
+                    ).as_json()["primaryConnectionString"]
+            for ep in endpoints["serviceBusQueues"]:
+                if ep.get("connectionString"):
+                    endpoint_props = parse_iot_hub_message_endpoint_connection_string(ep["connectionString"])
+                    namespace = endpoint_props["Endpoint"].strip("sb://").split(".")[0]
+                    ep["connectionString"] = cli.invoke(
+                        "servicebus queue authorization-rule keys list --namespace-name {} --resource-group {} "
+                        "--queue-name {} --name {}".format(
+                            namespace,
+                            ep["resourceGroup"],
+                            endpoint_props["EntityPath"],
+                            endpoint_props["SharedAccessKeyName"]
+                        )
+                    ).as_json()["primaryConnectionString"]
+            for ep in endpoints["serviceBusTopics"]:
+                if ep.get("connectionString"):
+                    endpoint_props = parse_iot_hub_message_endpoint_connection_string(ep["connectionString"])
+                    namespace = endpoint_props["Endpoint"].strip("sb://").split(".")[0]
+                    ep["connectionString"] = cli.invoke(
+                        "servicebus topic authorization-rule keys list --namespace-name {} --resource-group {} "
+                        "--topic-name {} --name {}".format(
+                            namespace,
+                            ep["resourceGroup"],
+                            endpoint_props["EntityPath"],
+                            endpoint_props["SharedAccessKeyName"]
+                        )
+                    ).as_json()["primaryConnectionString"]
+            for ep in endpoints["storageContainers"]:
+                if ep.get("connectionString"):
+                    endpoint_props = parse_storage_container_connection_string(ep["connectionString"])
+                    ep["connectionString"] = cli.invoke(
+                        "storage account show-connection-string -n {} -g {}".format(
+                            endpoint_props["AccountName"],
+                            ep["resourceGroup"],
+                        )
+                    ).as_json()["connectionString"]
+
+            file_upload = hub_resource["properties"]["storageEndpoints"]["$default"]
+            if file_upload["connectionString"]:
+                endpoint_props = parse_storage_container_connection_string(file_upload["connectionString"])
+                file_upload["connectionString"] = cli.invoke(
+                    "storage account show-connection-string -n {}".format(
+                        endpoint_props["AccountName"]
+                    )
+                ).as_json()["connectionString"]
+            print("Saved ARM template.")
+
+        return hub_state
+
+    def upload_hub_from_dict(self, hub_state: dict, hub_aspects: list = []):
```

Comment:
Please change the current mutable default value to None as described https://github.com/Azure/azure-iot-cli-extension/pull/582/files#r1003881435